backend/src/routes/api/navigate/group.js

```diff
@@ -21,6 +21,37 @@ export const getSimpleScene = async (sceneId) => {
   return scene;
 };
 
+const deleteNoteNoRoleCheck = async (noteId, groupId) => {
+  const note = await Note.findById(noteId);
+  const updateQuery = {
+    $pull: { [`notes.${note.role}`]: noteId },
+  };
+  //  delete note from group
+  await Group.updateOne({ _id: groupId }, updateQuery);
+
+  if (!note) {
+    throw new HttpError("Note not found", STATUS.NOT_FOUND);
+  }
+  await note.delete();
+  return true;
+};
+
+const deleteAllNotes = async (groupData) => {
+  const groupId = groupData._id;
+  const group = await Group.findById(groupId, { notes: 1 }).lean();
+  if (!group) {
+    throw new HttpError("Group not found", STATUS.NOT_FOUND);
+  }
+  const noteList = Object.entries(group.notes).flatMap(([role, ids]) =>
+    ids.map((id) => ({ role, id }))
+  );
+  if (noteList && noteList.length > 0) {
+    await Promise.all(
```

Comment:
We could simplify this quite a bit (and probably make it more optimised by using a batch delete operation. HINT: See mongoose `deleteMany`.